pysmt/typing.py

```diff
@@ -204,6 +209,30 @@ def is_array_type(self):
 # EOC _ArrayType
 
 
+class _RegExType(PySMTType):
+    """Internal class used to represent an Regex type.
+
+    This class should not be instantiated directly, but the factory
+    method RegExType should be used instead.
+    """
+    def __init__(self, regex_type):
+        decl = _TypeDecl("RegEx", 1)
```

Comment:
The definition of the RegLan sort has arity 0, not 1:  RegLan() and RegEx(String) are the same thing. This simplifies the handling of the type in this module, and it can be treated as other simple types like INT or REAL.